### DISCUSSION TITLE
Git submodules support

### DISCUSSION THREAD

**Discussion by silasb** [community/community#3493] [Codespaces] [answered]
I'm working with a repo that has submodule support and can't seem to commit to the submodule repo from within a codespaces.

```
git push -u origin cpu-features-by-ifndefs
remote: Permission to silasb/ChibiOS-Contrib.git denied to silasb.
fatal: unable to access 'https://github.com/silasb/ChibiOS-Contrib/': The requested URL returned error: 403
```

is the error that I'm getting.

**Comment by bartr** [comment, marked as answer]
I suspect you're running into the same issue that @2percentsilk  posted about in another thread. I don't know how to link her answer, so cross-posting. 

By default, we scope permissions to the repository your codespace is built off-of so a side effect is that you do not have permissions <to other repos>.

However, I have a workaround for you in the meantime 😄

Create a Personal Access Token with the following permissions:
public_repo
read:discussion
read:org

Depending on the permissions of the upstream repo, you may need additional permissions but the resulting terminal error should let you know which ones!

Set the environment variable GITHUB_TOKEN to equal your PAT.

You can do this by running export GITHUB_TOKEN=$PAT in your terminal.

< I added this line to my .bashrc via dotfiles, so it's automatic - works like a champ >

Let me know how that goes and rest assured, we're working on streamlining this and hope to share it with you soon 🎉

**Reply by mdehling** [reply]
Any updates on this?  I'm using this workaround (thanks!) but it would be nice to have a better solution eventually :-)
